libs/isograph-react/src/core/FragmentReference.ts

```diff
@@ -35,7 +35,7 @@ export type FragmentReference<
 export function stableIdForFragmentReference(
   fragmentReference: FragmentReference<any, any>,
 ): string {
-  return `${fragmentReference.root.__link}/TODO_FRAGMENT_NAME/${serializeVariables(fragmentReference.variables ?? {})}`;
+  return `${fragmentReference.root.__link}/${fragmentReference.root.__typename}/TODO_FRAGMENT_NAME/${serializeVariables(fragmentReference.variables ?? {})}`;
```

Comment:
nit: let's put the typename before the `__link` here, since that reflects how the store is organized